servers/src/main/java/tachyon/worker/TachyonWorker.java

```diff
@@ -15,20 +15,111 @@
 
 package tachyon.worker;
 
+import java.io.IOException;
+import java.net.InetSocketAddress;
+
+import org.apache.thrift.protocol.TBinaryProtocol;
+import org.apache.thrift.server.TThreadPoolServer;
+import org.apache.thrift.transport.TServerSocket;
+import org.apache.thrift.transport.TTransportException;
+import org.apache.thrift.transport.TTransportFactory;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.google.common.base.Throwables;
+
 import tachyon.Constants;
+import tachyon.Version;
+import tachyon.conf.TachyonConf;
+import tachyon.metrics.MetricsSystem;
+import tachyon.security.authentication.AuthenticationUtils;
+import tachyon.thrift.BlockWorkerClientService;
+import tachyon.thrift.WorkerNetAddress;
+import tachyon.util.network.NetworkAddressUtils;
+import tachyon.util.network.NetworkAddressUtils.ServiceType;
+import tachyon.web.UIWebServer;
+import tachyon.web.WorkerUIWebServer;
 import tachyon.worker.block.BlockWorker;
+import tachyon.worker.block.BlockWorkerClientServiceHandler;
 import tachyon.worker.file.FileSystemWorker;
 
 /**
  * Entry point for the Tachyon Worker. This class is responsible for initializing the different
  * workers that are configured to run.
+ * <p>
+ * This class is not thread-safe.
  */
 public final class TachyonWorker {
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
+  private TachyonConf mTachyonConf;
+  // The workers
+  /** The worker serving blocks */
+  private BlockWorker mBlockWorker;
+  /** The worker serving file system operations */
+  private FileSystemWorker mFileSystemWorker;
+
+  /** is true if the worker is serving the RPC server */
+  private boolean mIsServingRPC = false;
+  /** Worker metrics system */
+  private MetricsSystem mWorkerMetricsSystem;
+  /** Worker Web UI server */
+  private UIWebServer mWebServer;
+  /** Thread pool for thrift */
+  private TThreadPoolServer mThriftServer;
+  /** Server socket for thrift */
+  private TServerSocket mThriftServerSocket;
+  /** RPC local port for thrift */
+  private int mRPCPort;
+  /** The address for the rpc server */
+  private InetSocketAddress mWorkerAddress;
+  /** Net address of this worker */
+  private WorkerNetAddress mWorkerNetAddress;
+  /** Worker start time in milliseconds */
+  private long mStartTimeMs;
+
+  /**
+   * Constructor of {@link TachyonWorker}.
+   */
+  public TachyonWorker() {
+    try {
+      mStartTimeMs = System.currentTimeMillis();
+      mTachyonConf = WorkerContext.getConf();
+
+      mBlockWorker = new BlockWorker();
+      // Setup the file worker
```

Comment:
either remove this comment or move it one line up and change it to `Setup individual workers`